src/ADAL.PCL.Android/PlatformInformation.cs

```diff
@@ -63,5 +67,20 @@ public override string GetDeviceModel()
         {
             return null;
         }
+
+        public override string GetDeviceId()
+        {
+            return Settings.Secure.AndroidId.ToString();
+        }
+
+        public override string GetApplicationName()
+        {
+            return Application.Context.PackageManager.GetPackageInfo(Application.Context.PackageName, 0).ApplicationInfo.Name.ToString();
```

Comment:
I think we should consider using Package name instead of Application name. Application name can change but it is harder to change packagename. Please speak to @weijjia and @addev-ashish about this.